src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,14 +130,14 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
+            } else {
+                // read timestamp into DateTime object - the formatter delivers a timestamp
+                $dateTime = new \DateTime(sprintf('@%s', $timestamp));
+                // set timezone separately, as it would be ignored if set via the constructor,
+                // see http://php.net/manual/en/datetime.construct.php
+                $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
```

Comment:
@nicolas-grekas @HeahDude I'm sorry to bother you with such trivial change but I really don't understand why this line should be moved outside the else. 
The line 139 and the comment line 137 and 138 only applies to the construct of line 136 because it's created with the '@' with a timestamp.

Line 133 the timezone is already zpplied when constructed.

If this something about limiting the number of instruction in the `else`?